lib/internal/process/esm_loader.js

```diff
@@ -60,17 +61,26 @@ async function initializeLoader() {
     cwd = 'file:///';
   }
 
+  const keyedExportsList = [];
+
   // A separate loader instance is necessary to avoid cross-contamination
   // between internal Node.js and userland. For example, a module with internal
   // state (such as a counter) should be independent.
   const internalEsmLoader = new ESMLoader();
 
-  // Importation must be handled by internal loader to avoid poluting userland
-  const keyedExportsList = await internalEsmLoader.import(
-    customLoaders,
-    pathToFileURL(cwd).href,
-    ObjectCreate(null),
-  );
+  for (let i = 0; i < customLoaders.length; i++) {
+    const customLoader = customLoaders[i];
+
+    // Importation must be handled by internal loader to avoid poluting userland
+    const keyedExportsSublist = await internalEsmLoader.import(
+      [customLoader],
+      pathToFileURL(cwd).href,
+      ObjectCreate(null),
+    );
+
+    await internalEsmLoader.addCustomLoaders(keyedExportsSublist);
+    ArrayPrototypePushApply(keyedExportsList, keyedExportsSublist);
+  }
```

Comment:
A) How is implementing “loaders applying to subsequent loaders” so simple, and B) does this break any existing tests?

Looking at this code, there’s no way I would’ve guessed that this was using existing loaders as part of resolving and loading subsequent loaders. Could you add a comment (or several) explaining that that’s what’s happening here, and how?

Reply:
My naive thinking is that since loader resolution goes through a loader pipeline (even if it doesn't initially contain any loader but the default one), all we need is to progressively add the loaders to it so that they are used by subsequent resolutions.

A few tests don't seem to pass yet, I'm searching why.